app/Providers/AppServiceProvider.php

```diff
@@ -117,6 +118,22 @@ protected function registerCustomValidators()
 			// Useful for textareas - accepts letters, comas, dots, spaces and hyphens
 			return preg_match('/^[\pL\s\d-,.:;()""]+$/u', $value);
 		});
+
+		Validator::extend('morph_exists', function ($attribute, $value, $parameters, $validator) {
+			if (!$type = array_get($validator->getData(), $parameters[0], false)) {
+				return false;
+			}
+
+			if (Relation::getMorphedModel($type)) {
+				$type = Relation::getMorphedModel($type);
+			}
+
+			if (!class_exists($type)) {
+				return false;
+			}
+
+			return !empty(resolve($type)->find($value));
```

Comment:
Would that work? No need to get the whole row from DB.
```suggestion
			return resolve($type)->whereKey($value)->exists();
```